controllers/managedmcg_controller.go

```diff
@@ -697,6 +704,31 @@ func (r *ManagedMCGReconciler) updateComponentStatus() {
 		r.Log.Info("error getting Alertmanager, setting component status to Unknown")
 		amStatus.State = mcgv1alpha1.ComponentUnknown
 	}
+
+	// Getting the status of the Console component.
+	r.Log.Info("updating the Console component status")
+	consoleStatus := &r.managedMCG.Status.Components.Console
+	switch err := r.get(r.console); {
+	case err == nil:
+		consoleDeployment := &appsv1.Deployment{}
+		consoleDeployment.Namespace = r.namespace
+		consoleDeployment.Name = mcgmsConsoleName
+		if err := r.get(consoleDeployment); err == nil {
+			desiredReplicas := int32(1)
+			if consoleDeployment.Status.ReadyReplicas < desiredReplicas {
+				consoleStatus.State = mcgv1alpha1.ComponentPending
+			} else {
+				consoleStatus.State = mcgv1alpha1.ComponentReady
+			}
+		} else {
+			consoleStatus.State = mcgv1alpha1.ComponentPending
+		}
+	case errors.IsNotFound(err):
+		consoleStatus.State = mcgv1alpha1.ComponentNotFound
+	default:
+		r.Log.Info("error getting Console, setting component status to unknown")
+		consoleStatus.State = mcgv1alpha1.ComponentUnknown
+	}
 }
```

Comment:
I'd suggest breaking off the Noobaa, Alertmanager, Prometheus, and Console status updation parts into their own functions, and calling them here to reduce the cognitive complexity: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/red-hat-storage_mcg-osd-deployer/76/pull-ci-red-hat-storage-mcg-osd-deployer-main-lint/1556936379416449024#1:build-log.txt%3A24.